lib/middleware/proxy_to_reflection.coffee

```diff
@@ -3,11 +3,12 @@
 # https://github.com/artsy/reflection
 #
 httpProxy = require 'http-proxy'
+request = require 'superagent'
 proxy = httpProxy.createProxyServer()
 { REFLECTION_URL } = process.env
 
 module.exports = (req, res, next) ->
-  if req.query._escaped_fragment_?
-    proxy.web req, res, target: REFLECTION_URL, changeOrigin: true
-  else
-    next()
+  return next() unless req.query._escaped_fragment_?
+  request.head(REFLECTION_URL + req.url).end (err) ->
+    return next() if err
+    proxy.web req, res, target: REFLECTION_URL, changeOrigin: true
```

Comment:
Yeah, [Node HTTP Proxy](https://github.com/nodejitsu/node-http-proxy) works by inspecting the `req` object and proxying _any_ requests from the provided base url. The `request.head(REFLECTION_URL + req.url)` above is using a different lib that makes the request directly and hence needs to concatenate the base and path.